Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -442,6 +442,8 @@ internal struct RedirectHandler<T> {
 
         var request = self.request
         request.url = redirectURL
+        request.host = redirectURL.host!
```

Comment:
host and scheme can be safely unwrapped as they have already been validated earlier

Reply:
imo we should be defensive here, so we don't make assumption about the call site which could change over time

Reply:
@tomerd I was thinking that maybe host and scheme should be resolved on bootstrap connect, that way host and scheme could be optional and there would be no need to have throwing `HTTPClient.Request` init? Most helper methods accept string as an URL, so error will be inside returned future anyway... What do you think?

Reply:
+1 on non-throwing `HTTPClient.Request` ctor

Reply:
this should probably be done in separate PR to reduce scope, maybe create an issue?

Reply:
@artemredkin @tomerd Does this mean I can leave this as it is for now? If yes, I have fixed all reviews and the PR should now be mergeable.

Reply:
@adtrevor how about you add a check here until @artemredkin takes on the bigger change? can be a simple as 

```
if let host = redirectURL.host {
  request.host = host
}
```

Reply:
@tomerd Thanks! Done and added an `assertionFailure` on else so that it doesn't silently fail on debug as it's still something unexpected.

Reply:
Thank you, LGTM! One small comment though, I think `preconditionFailure` would be more appropriate here, imo (we use it in other parts of the codebase as well).

Reply:
@artemredkin Replaced it! Thanks!